IoT-Sample/Swift/IoTSampleSwift/ConnectionViewController.swift

```diff
@@ -19,207 +19,273 @@ import AWSMobileClient
-                        let defaults = UserDefaults.standard
-                        let certificateId = defaults.string( forKey: "certificateId")
-
-                        self.logTextView.text = "Using certificate:\n\(certificateId!)\n\n\nClient ID:\n\(uuid)"
-
-                        tabBarViewController.viewControllers = [ self, self.publishViewController, self.subscribeViewController ]
-
-                    case .disconnected:
-                        tabBarViewController.mqttStatus = "Disconnected"
-                        print( tabBarViewController.mqttStatus )
-                        self.activityIndicatorView.stopAnimating()
-                        self.logTextView.text = nil
-
-                    case .connectionRefused:
-                        tabBarViewController.mqttStatus = "Connection Refused"
-                        print( tabBarViewController.mqttStatus )
-                        self.activityIndicatorView.stopAnimating()
-                        self.logTextView.text = tabBarViewController.mqttStatus
-
-                    case .connectionError:
-                        tabBarViewController.mqttStatus = "Connection Error"
-                        print( tabBarViewController.mqttStatus )
-                        self.activityIndicatorView.stopAnimating()
-                        self.logTextView.text = tabBarViewController.mqttStatus
-
-                    case .protocolError:
-                        tabBarViewController.mqttStatus = "Protocol Error"
-                        print( tabBarViewController.mqttStatus )
-                        self.activityIndicatorView.stopAnimating()
-                        self.logTextView.text = tabBarViewController.mqttStatus
+                self.logTextView.text = "Connecting (data plane)..."
+            }
+            let uuid = UUID().uuidString
+            // Connect to the AWS IoT data plane service over websocket
+            iotDataManager.connectUsingWebSocket(withClientId: uuid, cleanSession: true, statusCallback: mqttEventCallbackWebsocket(_:))
 
-                    default:
-                        tabBarViewController.mqttStatus = "Unknown State"
-                        print("unknown state: \(status.rawValue)")
-                        self.activityIndicatorView.stopAnimating()
-                        self.logTextView.text = tabBarViewController.mqttStatus
-                }
-                
-                NotificationCenter.default.post( name: Notification.Name(rawValue: "connectionStatusChanged"), object: self )
+        } else {
+            handleDisconnect()
+            DispatchQueue.main.async {
+                sender.setTitle("Connect \(self.IOT_WEBSOCKET)", for:UIControl.State())
+                sender.isEnabled = true
             }
         }
 
-        if (connected == false)
-        {
+    }
+    
+    @objc func connectButtonPressed(_ sender: UIButton) {
+        sender.isEnabled = false
+        if (connected == false) {
```

Comment:
as elsewhere, prefer guard/let to big if/else blocks

Reply:
Moved connect logic to separate function to cut down on the length of the `if`.  Not using guard, I consider guard as a precondition.